docs/detections/api/rules/rules-api-bulk-actions.asciidoc

```diff
@@ -398,13 +398,49 @@ saved object ID of the action.
 
 |==============================================
 
+Optional fields:
+
+[width="100%",options="header"]
+|==============================================
+|Name |Type |Description
+
+|frequency |String a|Object containing action’s frequency:
+
+* `summary` (Boolean, required): Defines whether notification will be sent for summary of alerts or for each generated alert individually.
+
+* `notifyWhen` (String, required`): Defines how often alerts generate actions. Valid values are:
+
+** `onActiveAlert`: Actions run when the alert becomes active and at each check interval while the rule conditions are met.
```

Comment:
@e40pud I'm not sure I understand this - when does an alert become active? Is this a concept that's specific to alerts generated by Security detection rules?

Reply:
I copied description of these values from the alerting framework API docs https://www.elastic.co/guide/en/kibana/current/create-rule-api.html

If I remember correctly, alerting framework addressing security solution rules as alerts. So, in our case we could say `rule becomes active`.

@marshallmain can you confirm that my understanding is correct here?

Reply:
Yeah the alerting framework has a more general description of `onActiveAlert` here that might be confusing to security users. The alerting framework has the capability to track an alert's status across multiple rule executions. E.g. if there's a rule checking for server CPU usage exceeding some threshold, it can keep an alert open and update the alert each time the rule runs and finds that the CPU usage is **still** exceeding the threshold. My understanding is the `at each check interval while the rule conditions are met.` part is intended to describe this behavior where it runs actions for that alert every time the rule runs as long as the CPU usage is still exceeding the threshold.

Alerts for security rules are more "fire and forget" where we don't re-check past alerts on the next rule execution, so actions won't keep firing "while rule conditions are met".

With that in mind, perhaps we can tailor the description here to the more specific security rule implementation and say something like "Actions run when the alert is generated" for `onActiveAlert` and "Actions run on the throttle interval and summarize new alerts from that interval" for `onThrottleInterval`.

Reply:
```suggestion
** `onActiveAlert`: Actions run when the alert is generated.
```